app/src/main/java/nl/nfi/cellscanner/MainActivity.java

```diff
@@ -126,19 +129,24 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
      * Export data via email
      */
     public void exportData(View view) {
-        if (!Database.getDataFile(this).exists()) Toast.makeText(getApplicationContext(), "No database present.", Toast.LENGTH_SHORT).show();
-        else {
-            Uri contentUri = getUriForFile(this, ".fileprovider", Database.getDataFile(this));
-            Intent sharingIntent = new Intent(Intent.ACTION_SENDTO);
+        if (!Database.getDataFile(this).exists()){
+            Toast.makeText(getApplicationContext(), "No database present.", Toast.LENGTH_SHORT).show();
 
-            sharingIntent.setData(Uri.parse("mailto:")); // only email apps should handle this
-//            sharingIntent.putExtra(Intent.EXTRA_EMAIL, "email to send to ");
-            sharingIntent.putExtra(Intent.EXTRA_SUBJECT, "datafile cell-scanner " +  getFileTitle());
-            sharingIntent.putExtra(Intent.EXTRA_STREAM, contentUri);
+        } else {
+            String[] TO = {""};
 
-            if (sharingIntent.resolveActivity(getPackageManager()) != null) {
-                startActivity(sharingIntent);
-            }
+            Uri uri = FileProvider.getUriForFile(getApplicationContext(), "nl.nfi.cellscanner.fileprovider", Database.getDataFile(getApplicationContext()));
+
+            Intent sharingIntent = new Intent(android.content.Intent.ACTION_SEND);
+
+            sharingIntent.putExtra(Intent.EXTRA_EMAIL, TO);
+            sharingIntent.putExtra(android.content.Intent.EXTRA_SUBJECT, getFileTitle());
+            sharingIntent.putExtra(Intent.EXTRA_STREAM, uri);
+
+            //need this to prompts email client only
+            sharingIntent.setDataAndType(uri, "message/rfc822");
+
+            startActivity(Intent.createChooser(sharingIntent, "Share via"));
```

Comment:
The sharing options went back from only email to all apps (android 7) and to gmail services (android 8, pixel 2 virtual device). Not a problem, since we want to change to an automated cloud service

Reply:
The other option resulted in not being able to attach the file =/